sys_conn_oob.go

```diff
@@ -325,7 +332,7 @@ func appendIPv6ECNMsg(b []byte, val protocol.ECN) []byte {
 	h.SetLen(unix.CmsgLen(dataLen))
 
 	// UnixRights uses the private `data` method, but I *think* this achieves the same goal.
-	offset := startLen + unix.CmsgSpace(0)
-	b[offset] = val.ToHeaderBits()
+	dataPtr := unsafe.Pointer(uintptr(unsafe.Pointer(h)) + uintptr(unix.CmsgSpace(0)))
+	*(*int32)(dataPtr) = int32(val.ToHeaderBits())
```

Comment:
Why is this using `unsafe`?

Reply:
I didn't really understand what happened to this, this logic was modified from x/net/quic, and seems working correctly.

Reply:
Why can't we just keep it as it was (maybe with a `NativeEndian` change, if needed)?

Reply:
I'm confused too, apparently `ecn` only needs to be set by two bits, which shouldn't be affected by byte order, but the bug does exist. I haven't really checked what the actual representation in memory, since `delve` is not available on the `s390x`.

Reply:
There's only two options, right? Big and little endian. You could try both, and observe what's sent out on the wire using Wireshark.

Reply:
This is not the kind of error that would be captured by Wireshark; the problem lies in the parameters sent to the syscall, and I will try using strace to see if there are any changes.